src/Shared/Sdk/Providers/ServiceClients/ClientOptions.cs

```diff
@@ -0,0 +1,12 @@
+namespace ACMTTU.NoteSharing.Shared.SDK.Clients
+{
+    /// EnvironmentVariableOptions
+    /// 
+    /// Determines what environment variables are available
+    /// for services in Platform to consume
+    public enum ClientOptions
```

Comment:
I'm generally not a fan of having an enum of a set of options like this, and would instead like to see BlobStorage and Database be classes polymorphic to one another, and both inherit from the same base. This would clean up the case statements above, as well as allow easy mocking for testing purposes. It also means new ClientOptions can be created without having to keep a running list of all options up to date.

Reply:
That was my first implementation actually so I 100% agree with you. I would rather have two different types of client factories. One for the CosmosDB account and one for the Storage account.

The reason why you don't see it on here, is because I would then have one method...

```csharp
public ClientType GetClient<ClientType>(string connectionString) {
  return Client
}
```

I don't know how to return either a CosmosDB or a Storage client from a method, much less create an interface for it.

I'm starting to think...

Do you think I'm trying to do too much with this class lib?

Right now, I am getting the connection string, as well as creating a Client. Do you think I should leave it at grabbing the connection string and leave the creation of the Client itself to the users?

Reply:
well actually... hold on lemme work on this real quick. But I do agree with you.

Reply:
Look at the IHttpClientFactory: The method CreateClient() is the method required to return the constructed client. Then you would have

```chsarp
public static DBClient CreateClient(ClientType t, ConnectionString cs){
...
}
```
If you use the factory pattern you can leave fetching to the user and allow construction to the DI IoC container.